src/qt/optionsdialog.cpp

```diff
@@ -466,13 +466,13 @@ void OptionsDialog::updateDefaultProxyNets()
     bool has_proxy;
 
     has_proxy = model->node().getProxy(NET_IPV4, proxy);
-    ui->proxyReachIPv4->setChecked(has_proxy && proxy.ToString() == proxyIpText);
+    ui->proxyReachIPv4->setChecked((has_proxy && proxy.ToString() == proxyIpText) && g_reachable_nets.Contains(NET_IPV4));
```

Comment:
> Also, this change is mixing the "reachable nets" with the "which network is accessed via which proxy" which are two different things. Maybe drop this change altogether?

I get it, that's correct, but also when `-onlynet` is specified, g_reachable_nets contains only the nets that are specified by each `-onlynet` option (ie so in this case `-onlynet=ipv6`, ipv4 peers won't be reached by the proxy): 

https://github.com/bitcoin-core/gui/blob/225718eda89d65a7041c33e1994d3bf7bd71bbdd/src/init.cpp#L1391-L1399

> It is not that we reach peers via IPv4. We reach IPv4 peers via the proxy. 

I agree, happy to do it here or on a follow-up to get this merged quicker.

"Used for reaching peers of type: [ ] IPv4/IPv6/CJDNS [ ] Tor".
(combine IPv4, IPv6 and CJDNS peers in one checkbox because we always have a single proxy for all those 3 networks, i.e. can't split it to have proxy1 for IPv4 and proxy2 for IPv6).

Also when a separate SOCKS5 for Tor is set, the "[ ] Tor" should disappear from the proxy above.

Reply:
I'll drop the changes on `src/qt/optionsdialog.cpp` as suggested by @vasild, since the approach is incorrect.